### ISSUE TITLE
remove function argument assist

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-analyzer/rust-analyzer#3254]
IMO it would be useful if I could select a function argument of any function inside the definition, mark it for removal, and RA removed the argument and the respective arguments of all call sites.

````rust
fn main() {
    println!("Hello, world!");
    function(0, 1);
}

fn function(a: u8, ┃b: u8) {
    unimplemented!();
}
````

=>

````rust
fn main() {
    println!("Hello, world!");
    function(0);
}

fn function(a: u8) {
    unimplemented!();
}
````

**Comment by lnicola** [comment]
The structural search and replace feature should be able to do this.

**Comment by matklad** [comment]
Heh, couple of days ago I was thinking about writing down the same issue for struct fields, but didn't get to it!

Thanks for the report!

I think this should be doable without our current state of affairs, although it'll probably required a few moving parts. Essentially, it should build on top of find_references infrasturucutre. 


But yeah, SSR is a way to approximate this feature today, if you are willing to match on the function call *just* by name of the function (ie, false positive when two functions are named the same).

**Comment by flodiebold** [comment]
I like the idea, but maybe the assist should only trigger if the argument is unused within the function (which we can also (approximately) detect with find_references).

**Comment by theotherphil** [comment]
I’d like to give this a go. Any pointers beyond just reading https://github.com/rust-analyzer/rust-analyzer/blob/0e5fe4715360ab2f0d2954c9a8b37b247335bbcc/crates/ide/src/references.rs and finding existing uses of these functions?

**Comment by Veykril** [comment]
I think this is already implemented unless I'm missing something https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/assists/src/handlers/remove_unused_param.rs

**Comment by theotherphil** [comment]
I think that’s missing the “arguments of all call sites” bit.

**Comment by theotherphil** [comment]
Implemented by https://github.com/rust-analyzer/rust-analyzer/pull/5821
